doc/architecture/trace_file_formats.adoc

```diff
@@ -16,9 +19,187 @@ The length does not include the integer itself.
+The fully qualified name to the message within the OSI descriptor set. For example, for the `SensorView` message type, this would be `osi3.SensorView`.
+
+`encoding`::
+The value `protobuf`.
+
+`data`::
+A binary FileDescriptorSet as produced by `protoc --include_imports --descriptor_set_out`.
+
+The schema record must be stored in the summary section of the trace file, and must be referenced by the OSI channels that use the schema.
+
+[#sec-osi-trace-file-multi-channel]
+=== OSI channel
+
+An OSI channel is a data stream within the OSI multi channel trace file that contains serialized OSI messages of the same type.
+Each OSI channel must be described by a channel record in the summary section of the trace file with the following fields:
+
+`id`::
+A file-wide unique identifier for the channel.
+
+`schema_id`::
+The ID of the schema record that describes the message type of the channel.
+
+`topic`::
+A unique name for the channel within the trace file.
+When recording OSI traces for a model packaged according to the OSI Sensor Model Packaging (OSMP) layer, using the naming conventions defined in the OSMP specification for variables as topics is recommended.
+For example, for a sensor model with two SensorView inputs and one SensorData output, the topic names would be `OSMPSensorViewIn[1]`, `OSMPSensorViewIn[2]`, and `OSMPSensorDataOut`, accordingly.
+In other cases, the topic name should be chosen to reflect the purpose of the channel, and should include some indication of the message type.
+
+`message_encoding`::
+The value `protobuf`.
+
+`metadata`::
+A map of additional meta-data for the channel.
+This map may contain arbitrary key-value pairs, however keys starting with `net.asam.osi` are reserved for use by {THIS_STANDARD}.
+It is strongly recommended to follow reverse domain name notation for custom meta-data keys to avoid conflicts.
+The following mandatory entries are defined by {THIS_STANDARD}:
+
+`net.asam.osi.trace.channel.osi_version`:::
+The version of the OSI schema used in creating the data of this OSI channel.
+This is a string in the format `major.minor.patch`, for example `3.0.0`.
+
+`net.asam.osi.trace.channel.protobuf_version`:::
+The version of the protobuf implementation used in creating the data of this OSI channel.
+This is a string in the format `major.minor.patch`, for example `3.17.3`.
+
+The following recommended entries are defined by {THIS_STANDARD}:
+
+`net.asam.osi.trace.channel.description`:::
+A human-readable description of the channel.
+
+The channel record must be stored in the summary section of the trace file, and must be referenced by the OSI message records that are part of the channel.
+
+All messages in an OSI channel must be stored in chunk records in the data section of the trace file.
+
+Each message record in a chunk record must contain the following fields:
+
+`channel_id`::
+The ID of the channel that the message belongs to.
+
+`publish_time`::
+The point in time when the message was published.
```

Comment:
In #833 we defined the publish_time as:

```
Must reflect the timestamp of the stored OSI top-level message
```

This makes it easier to use the indexing to access certain OSI time frames.

Reply:
> In #833 we defined the publish_time as:

Actually this was a copy and paste left over, see actual publish_time definition below (will remove this old duplicate).
> 
> ```
> Must reflect the timestamp of the stored OSI top-level message
> ```
> 
> This makes it easier to use the indexing to access certain OSI time frames.

It sadly does not, as solely `log_time` is used for indexing purposes in MCAP, see contortions below with `log_time` to deal with that.

Reply:
Yes, but if we want to put the same values in publish_time and log_time, I propose to make them both identical with the timestamp of the message.

Reply:
That's what the current definition does, with special allowance to still use log_time as intended if there is a - perceived - need.